Sources/Datadog/URLSessionAutoInstrumentation/Interception/URLSessionInterceptor.swift

```diff
@@ -30,66 +30,87 @@ internal protocol URLSessionInterceptorType: class {
 }
 
 internal class URLSessionInterceptor: URLSessionInterceptorType {
-    private let firstPartyHostsFilter: URLFilter
-    private let queue: DispatchQueue
+    /// Filters first party `URLs` defined by the user.
+    private let firstPartyURLsFilter: FirstPartyURLsFilter
+    /// Filters internal `URLs` used by the SDK.
+    private let internalURLsFilter: InternalURLsFilter
```

Comment:
i think we could keep those 2 merged in 1 `URLFilter`, is there a reason for the split?

Reply:
I did the split, as both work very differently and are configured with distinct information. It's more SRP to have two, instead of one `URLFilter` which would have half code dedicated for "first party hosts filtering" and second half for "internal URLs filtering", but nothing in common (the [cohesion](https://en.wikipedia.org/wiki/Cohesion_(computer_science)) score for such component would be `0`).